File: spec/factories/saml_service_providers.rb
# frozen_string_literal: true

FactoryBot.define do
  factory :saml_service_provider do
    sequence(:name) { |n| "Test Service Provider #{n}" }
    sequence(:entity_id) { |n| "http://localhost:3000/users/saml/metadata/#{n}" }
    acs_urls { ['http://localhost:3000/users/saml/auth'] }
    enabled { true }
    mask_identity { false }
    require_signed_requests { false }
    association :project
  end
end

File: app/frontend/modules/admin/modules/client/core/samlServiceProviders.ts
import * as t from 'io-ts'

export const SamlServiceProviderTR = t.type({
  id: t.string,
  name: t.string,
  entityId: t.string,
  acsUrls: t.array(t.string),
  idpCertificate: t.string,
  ssoServiceUrl: t.string,
  issuerUri: t.string,
  projectId: t.number,
  enabled: t.boolean,
  maskIdentity: t.boolean,
  createdAt: t.string,
  updatedAt: t.string,
})

export type SamlServiceProvider = t.TypeOf<typeof SamlServiceProviderTR>
